test/unit/util/GuardedStream.test.ts

```diff
@@ -118,24 +117,18 @@ describe('GuardedStream', (): void => {
       const errorCb2 = jest.fn();
       stream.on('error', errorCb);
       stream.on('error', errorCb2);
-      expect(stream.listenerCount('error')).toBe(2);
-      expect(stream.listenerCount('newListener')).toBe(0);
-      expect(stream.listenerCount('removeListener')).toBe(1);
+      expect(stream.listenerCount('error')).toBe(3);
+      expect(stream.listenerCount('newListener')).toBe(1);
       stream.removeListener('error', errorCb2);
-      expect(stream.listenerCount('error')).toBe(1);
-      expect(stream.listenerCount('newListener')).toBe(0);
-      expect(stream.listenerCount('removeListener')).toBe(1);
+      expect(stream.listenerCount('error')).toBe(2);
+      expect(stream.listenerCount('newListener')).toBe(1);
       stream.removeListener('error', errorCb);
       expect(stream.listenerCount('error')).toBe(1);
       expect(stream.listenerCount('newListener')).toBe(1);
-      expect(stream.listenerCount('removeListener')).toBe(0);
 
       stream.emit('error', new Error('error'));
 
-      const listen = new Promise((resolve, reject): void => {
-        stream.on('end', resolve);
-        stream.on('error', reject);
-      });
+      const listen = getListenPromise(stream);
```

Comment:
And here, we can simply replace it by attaching a `jest.fn()` to error and checking if it was called (as above).